### PR TITLE
Update documentation to stateless components

### PR DESCRIPTION
**Pull request by mxstbr**
See #5949, as @jimfb recommended I updated two pages to stateless components. For now I've only done the english ones, if this is fine I'll update the translations as well.

### CODE REVIEW

File: docs/tips/14-communicate-between-components.md [hunk 1]
```diff
@@ -13,23 +13,21 @@ For child-parent communication:
...
-      <div>
-        {this.props.items.map(function(item, i) {
-          return (
-            <div onClick={this.handleClick.bind(this, i)} key={i}>{item}</div>
-          );
-        }, this)}
-      </div>
-    );
+function GroceryList(props) {
+  var handleClick = function(i) {
```
Comment (jimfb) [inline docs/tips/14-communicate-between-components.md]: We can move this up out of the GroceryList component/function, and have it accept both props and index.  That saves us an allocation on each render, which reduces GC pressure, and is therefore probably a better practice.

**Comment by jimfb** [comment]
Sounds good.  One nitpick above.  Also, as you mentioned, would be good to update the translated docs too.  Then I think we're good-to-go, assuming there is no other feedback from others.

**Comment by mxstbr** [comment]
@jimfb nice one with the `handleClick`, makes sense. Updated the PR with the fix and the translations changed as well!

File: docs/docs/06-transferring-props.it-IT.md [hunk 2]
```diff
@@ -27,19 +27,17 @@ Nel resto di questo tutorial vengono illustrate le best practices, usando JSX e
...
+  var fancyClass = props.checked ? 'FancyChecked' : 'FancyUnchecked';
+  return (
+    <div className={fancyClass} onClick={props.onClick}>
+      {props.children}
+    </div>
+  );
+}
 ReactDOM.render(
   <FancyCheckbox checked={true} onClick={console.log.bind(console)}>
-    Ciao mondo!
```
Comment (jimfb) [inline docs/docs/06-transferring-props.it-IT.md]: Translation lost?

File: docs/docs/06-transferring-props.ko-KR.md [hunk 3]
```diff
@@ -27,19 +27,17 @@ React.createElement(Component, Object.assign({}, this.props, { more: 'values' })
...
+  var fancyClass = props.checked ? 'FancyChecked' : 'FancyUnchecked';
+  return (
+    <div className={fancyClass} onClick={props.onClick}>
+      {props.children}
+    </div>
+  );
+}
 ReactDOM.render(
   <FancyCheckbox checked={true} onClick={console.log.bind(console)}>
-    세상아 안녕!
```
Comment (jimfb) [inline docs/docs/06-transferring-props.ko-KR.md]: lost translation.

**Comment by mxstbr** [comment]
Ah damn yeah I missed the translations, one second.

File: docs/tips/14-communicate-between-components.ko-KR.md [hunk 4]
```diff
@@ -13,26 +13,24 @@ next: expose-component-functions-ko-KR.html
 `GroceryList` 컴포넌트가 배열로 생성된 아이템 목록을 가지고 있다고 해봅시다. 목록의 아이템이 클릭되면 아이템의 이름이 보이길 원할 겁니다:
 
 ```js
-var GroceryList = React.createClass({
-  handleClick: function(i) {
-    console.log('클릭한 아이템: ' + this.props.items[i]);
-  },
+var handleClick = function(i, props) {
+  console.log('You clicked: ' + items[i]);
```
Comment (jimfb) [inline docs/tips/14-communicate-between-components.ko-KR.md]: props.items?  I think items is undefined.

**Comment by mxstbr** [comment]
That's what I get for copy and pasting, fixed all the translations! Sorry about that!

File: docs/tips/14-communicate-between-components.ko-KR.md [hunk 5]
```diff
@@ -13,23 +13,21 @@ next: expose-component-functions-ko-KR.html
 `GroceryList` 컴포넌트가 배열로 생성된 아이템 목록을 가지고 있다고 해봅시다. 목록의 아이템이 클릭되면 아이템의 이름이 보이길 원할 겁니다:
 
 ```js
-var GroceryList = React.createClass({
-  handleClick: function(i) {
-    console.log('클릭한 아이템: ' + this.props.items[i]);
-  },
+var handleClick = function(i, props) {
+  console.log('클릭한 아이템: ' + items[i]);
```
Comment (jimfb) [inline docs/tips/14-communicate-between-components.ko-KR.md]: `props.items`?  I think items is undefined here, right?

**Comment by mxstbr** [comment]
Fixed

**Comment by jimfb** [comment]
Thanks!

**Pull request merged**